flake.nix

```diff
@@ -41,6 +41,19 @@
         };
         inherit (pkgs) lib;
 
+        # GHC versions to provide development environments for.
+        # Keep this synced with `./test-harness-macro/src/lib.rs`.
```

Comment:
I don't love that this needs to stay synced up with a list in a different format in the macro code:
```rust
const GHC_VERSIONS: [&str; 4] = ["9.0.2", "9.2.8", "9.4.6", "9.6.2"];
```
I like that specifying `ghc90` rather than `ghc902` uses the latest patch version in each of the GHC releases, but I'm not sure if there's a good way to go from there to a list of full versions.

Maybe we could set an environment variable to `lib.mkBinPath (builtins.map (drv: drv.version) ghcBuildInputs)` and then use that as the list of version numbers in the macro code?

Reply:
OK, I implemented this approach -- let me know what you think.